### PR TITLE
feat(cluster): add cluster options resource

### PR DESCRIPTION
**Pull request by rafsaf**
Adding cluster options resource

3 things:
- didn't use nested object in schema for options like "migration" but everything is in single mapping, if there is a need, some options can be moved to separate nested mappings
- didn't implement all options in the resource like tags-related 3 options, not sure how if this is ok (in API there is support for every field I think)
- in schema there are no strict validators, i found proxmox api quite ok describing what's wrong with params, this is also opinionated and can be changed.

### Contributor's Note
Please mark the following items with an [x] if they apply to your PR.
Leave the [ ] if they are not applicable, or if you have not completed the item.
- [x] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [x] I have added / updated templates in `/examples` for any new or updated resources / data sources.
- [x] I have ran `make examples` to verify that the change works as expected. 

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

### CODE REVIEW

**Comment by bpg** [comment]
Thanks a lot @rafsaf! I'll get to this PR in the next few days.
In a meantime, could you try and add some acceptance tests for the new resources? If not, it's fine too, I'll toss something in during the review.

**Comment by rafsaf** [comment]
Sure I will try.

File: internal/cluster/resource_options.go [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+
+	if iface.MaxWorkers != nil {
+		m.MaxWorkers = types.Int64Value(int64(*iface.MaxWorkers))
+	}
+
+	m.Console = types.StringPointerValue(iface.Console)
+	m.HTTPProxy = types.StringPointerValue(iface.HTTPProxy)
+	m.MacPrefix = types.StringPointerValue(iface.MacPrefix)
+	m.Description = types.StringPointerValue(iface.Description)
+	m.HAShutdownPolicy = types.StringPointerValue(iface.HASettings.ShutdownPolicy)
```
Comment (bpg-dev) [inline internal/cluster/resource_options.go]: `HASettings` can be `nil`, that will cause a panic here.
Same for `Migration`, `ClusterResourceScheduling` fields

File: internal/cluster/resource_options.go [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+// Schema defines the schema for the resource.
+func (r *clusterOptionsResource) Schema(
+	_ context.Context,
+	_ resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		Description: "Manages Proxmox VE Cluster Datacenter options.",
+		Attributes: map[string]schema.Attribute{
+			"id": structure.IDAttribute("A unique identifier"),
```
Comment (bpg-dev) [inline internal/cluster/resource_options.go]: minor, I think the default description provided by `IDAttribute()` is good enough, the optional override is for special cases, for example where ID has a special format that worth documenting, etc.

File: internal/cluster/resource_options.go [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/bpg/terraform-provider-proxmox/internal/structure"
+	"github.com/bpg/terraform-provider-proxmox/proxmox"
+	cluster "github.com/bpg/terraform-provider-proxmox/proxmox/cluster"
```
Comment (bpg-dev) [inline internal/cluster/resource_options.go]: minor, the import alias seems to be redundant

**Comment by rafsaf** [comment]
ok, probably should be fine now with some test file to run with TF_ACC.

i added some fixes due to crashes I discovered (nil pointer dereference :D) in some cases, maybe now this will be "good enough".

**Review by bpg — APPROVED** [review]
LGTM! 🚀

Thanks again for the PR, great work @rafsaf!

I've just added schema validators for the new attributes, and made a few other minor fixes.
I'll mere the PR tomorrow, in case you have any comments / updates.

**Comment by rafsaf** [comment]
I think it's fine, I will now have a clue how to write those validators (a bit frustrating in new language/software to dig how to do things properly, X times in a row). Thanks.

PS. Nice that you catch typo like "keybord", I literally copy pasted those texts from proxmox assuming it's correct :)

**Pull request merged**